pkg/chartutil/requirements.go

```diff
@@ -266,3 +268,128 @@ func ProcessRequirementsEnabled(c *chart.Chart, v *chart.Config) error {
 
 	return nil
 }
+
+// pathToMap creates a nested map given a YAML path in dot notation
+func pathToMap(path string, data map[string]interface{}) map[string]interface{} {
+	ap := strings.Split(path, ".")
+	n := []map[string]interface{}{}
```

Comment:
I suspect that's because you used the two-argument form of `make`, as opposed to the three-argument form, right? I meant the following:
```go
n := make([]map[string]interface{}, 0, len(ap))
```
If instead you use the two-argument ` make([]map[string]interface{}, len(ap))`, then you shouldn't use `append` later; instead, just assign to the slice elements by index, because it's already of the intended size.